pipelines/h4c/post_processing/v7_interleave/post_lstbin_frate/task_scripts/do_EXTRACT_AUTOS.sh

```diff
@@ -0,0 +1,36 @@
+#! /bin/bash
+set -e
+
+#import common functions
+src_dir="$(dirname "$0")"
+source ${src_dir}/_common.sh
+
+# This script extracts the autocorrelations into a single waterfall file
+label=${1}
+include_diffs=${2}
+
+if [ "${include_diffs}" = "true" ] 
+then
+    sumdiff=("sum" "diff")
+else
+    sumdiff=("sum")
+fi
+
+for sd in ${sumdiff[@]}
+do
+    # *Sigh* bash
+    # Use the return value of ls to see if it runs OK (don't want to pass an empty list for example)
+    # Redirect the output to die in /dev/null/ so it doesn't crowd the log
+    # Exit if there is an error.
+    if ls *${sd}.${label}.*chunked.uvh5 1> /dev/null 
+    then
+        flist=$(ls *${sd}.${label}.*chunked.uvh5)
+    else
+        echo "Error when calling ls on ${sd} files with label ${label}. Probably could not find files. Check error log. Exiting."
+        exit 1
```

Comment:
I'm a little confused about this, in the sense that I thought the "chunked" files would be the ones that have all of the times for a chunk of baselines and one spectral window, and so we wouldn't really want to look for those files&mdash;I would think that instead we would want to look for whatever the files look like fresh out of the LST binner.

Also, have you tested the magic on line 25? It's not obvious to me that it'll do the right thing (but I'm not a bash expert, so maybe it is obvious to someone with more experience).

Reply:
Files with all the times for a small number of baselines are called "waterfall" files, and these are not output in the unfiltered state. I need the unfiltered autos, because the FRF'd autos are not the mathematically correct item to use for error propagation. You have to do a slightly different, but related operation to the unfiltered autos to get the right covariances. 

The files fresh out of the LST-binner have 2 integrations each, but lots of baselines. The chunked files have fewer baselines but 20 integrations each. There is no processing being done between these as far as I know. 

I think it is less files to read in using the chunked files, so these are the files we should use for this step. 

Yeah, line 25 works. The exact meaning of everything is in some commented lines above the magic.

Reply:
Great, thanks for the explanation! My main concern is that I don't have a sense of whether we ought to expect the chunked files to have the autos in them, but I figure that's why you go through the trouble of checking the files for autos.